external/websocket/websocket.c

```diff
@@ -332,6 +332,10 @@ int websocket_client_handshake(websocket_t *client, char *host, char *port, char
 	}
 
 	memset(accept_key, 0, WEBSOCKET_ACCEPT_KEY_LEN);
+	if (keyhdend - keyhdstart > WEBSOCKET_CLIENT_KEY_LEN) {
+		WEBSOCKET_DEBUG("error key length\n");
+		goto EXIT_WEBSOCKET_HANDSHAKE_ERROR;
+	}
 	memcpy(accept_key, keyhdstart, keyhdend - keyhdstart);
 	websocket_create_accept_key(dst, WEBSOCKET_ACCEPT_KEY_LEN, client_key, WEBSOCKET_CLIENT_KEY_LEN);
 	accept_key[WEBSOCKET_ACCEPT_KEY_LEN - 1] = '\0';
```

Comment:
Is "WEBSOCKET_ACCEPT_KEY_LEN - 1" correct?
or
"accept_key[keyhdend - keyhdstart] = '\0';

Reply:
WEBSOCKET_ACCEPT_KEY_LEN - 1 is correct. keyhdstart and keyhdend are only the length of the header.